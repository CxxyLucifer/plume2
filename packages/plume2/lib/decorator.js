"use strict";
/**
 * 绑定Actor的Action
 * Usage:
 *  class HelloActor extends Actor {
 *     @Action('hello')
 *     hello(state) {
 *       return state;
 *     }
 *  }
 *
 * @param msg 事件名
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Action = function (msg) { return function (target, 
//@ts-ignore
property, descriptor) {
    target._route || (target._route = {});
    /**
     * 如果有actor的Action中有重名的事件名，warning
     */
    if (process.env.NODE_ENV != 'production') {
        if (target._route[msg]) {
            var actorName = target.constructor.name;
            console.warn("\uD83D\uDE0E" + actorName + " had @Action('" + msg + "'), Please review your code.");
        }
    }
    target._route[msg] = descriptor.value;
}; };
