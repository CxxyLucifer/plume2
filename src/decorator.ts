export function Action(msg: string): Function {
  return function(
    target: any,
    property: any,
    descriptor: TypedPropertyDescriptor<any>
  ) {
    target._route || (target._route = {});

    if (process.env.NODE_ENV != 'production') {
      if (target._route[msg]) {
        const actorName = target.constructor.name;
        console.warn(
          `ðŸ˜Ž${actorName} had @Action('${msg}'), Please review your code.`
        );
      }
    }

    target._route[msg] = descriptor.value;
  };
}
