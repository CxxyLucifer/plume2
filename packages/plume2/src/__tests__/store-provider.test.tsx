import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Actor, Store, StoreProvider } from '../index';
import { Action } from '../decorator';

class HelloActor extends Actor {
  defaultState() {
    return { name: 'plume' };
  }

  @Action('change')
  change(state) {
    return state.set('name', 'plume++');
  }
}

class AppStore extends Store {
  constructor(props) {
    super(props);
    window['_store'] = this;
  }

  bindActor() {
    return [new HelloActor()];
  }

  change = () => {
    this.dispatch('change');
  };
}

@StoreProvider(AppStore)
class Home extends React.Component {
  state: { name: string };

  render() {
    return <div>{this.state.name}</div>;
  }
}

describe('store provider test suite', () => {
  it('first render', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('store dispath change render', () => {
    const component = renderer.create(<Home />);
    const store = window['_store'] as AppStore;
    store.change();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('merge store state', () => {
    @StoreProvider(AppStore)
    class StateApp extends React.Component {
      constructor(props) {
        super(props);
        this.state = {
          text: 'merge store test'
        };
      }

      render() {
        expect(this.state).toEqual({
          text: 'merge store test',
          name: 'plume'
        });

        return <div />;
      }
    }

    const tree = renderer.create(<StateApp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
