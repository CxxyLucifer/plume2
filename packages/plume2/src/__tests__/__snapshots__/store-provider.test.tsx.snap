// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store provider test suite debug mode 1`] = `<div />`;

exports[`store provider test suite debug mode 2`] = `
Array [
  "%cplume2@1.1.0ðŸš€",
  "StoreProvider(DebugApp) will mount ðŸš€",
]
`;

exports[`store provider test suite first render 1`] = `
<div>
  plume
</div>
`;

exports[`store provider test suite merge store state 1`] = `<div />`;

exports[`store provider test suite store dispath change render 1`] = `
<div>
  plume++
</div>
`;
